logic/Rule.java

```diff
@@ -224,12 +221,12 @@ public String toString() {
     public static class Condition {
 
         private final Rule rule;
-        private Set<Concludable> concludablesTriggeringRules;
+        private final ResolvableConjunction conjunction;
         private Set<Concludable> negatedConcludablesTriggeringRules;
 
         Condition(Rule rule) {
             this.rule = rule;
-            this.concludablesTriggeringRules = null;
+            this.conjunction = ResolvableConjunction.of(rule.when());
```

Comment:
This is a bit awkward. Since Rule maintains ownership of when (the conjunction pattern), but Condition has owner ship of the resolvableConjunction corresponding to when. 
(Conclusion has the same problem)

Reply:
Outdated; This was in a different PR